packages/rest/src/router/route-sort.ts

```diff
@@ -28,41 +28,30 @@ export function compareRoute(
   route1: Pick<RouteEntry, 'verb' | 'path'>,
   route2: Pick<RouteEntry, 'verb' | 'path'>,
 ): number {
-  // First check verb
-  const verb1 = HTTP_VERBS[route1.verb.toLowerCase()] || HTTP_VERBS.get;
-  const verb2 = HTTP_VERBS[route2.verb.toLowerCase()] || HTTP_VERBS.get;
-  if (verb1 !== verb2) return verb1 - verb2;
-
-  // Then check the path tokens
+  // First check the path tokens
   const path1 = route1.path.replace(/{([^}]*)}(\/|$)/g, ':$1$2');
   const path2 = route2.path.replace(/{([^}]*)}(\/|$)/g, ':$1$2');
   const tokensForPath1: pathToRegExp.Token[] = parse(path1);
   const tokensForPath2: pathToRegExp.Token[] = parse(path2);
 
-  // Longer path comes before shorter path
-  if (tokensForPath1.length < tokensForPath2.length) {
-    return 1;
-  } else if (tokensForPath1.length > tokensForPath2.length) {
-    return -1;
-  }
+  const length =
+    tokensForPath1.length > tokensForPath2.length
+      ? tokensForPath1.length
+      : tokensForPath2.length;
 
-  // Now check token by token
-  for (let i = 0; i < tokensForPath1.length; i++) {
+  for (let i = 0; i < length; i++) {
     const token1 = tokensForPath1[i];
     const token2 = tokensForPath2[i];
-    if (typeof token1 === 'string' && typeof token2 === 'string') {
-      if (token1 < token2) return -1;
-      else if (token1 > token2) return 1;
-    } else if (typeof token1 === 'string' && typeof token2 === 'object') {
-      // token 1 is a literal while token 2 is a parameter
-      return -1;
-    } else if (typeof token1 === 'object' && typeof token2 === 'string') {
-      // token 1 is a parameter while token 2 is a literal
-      return 1;
-    } else {
-      // Both token are parameters. Treat as equal weight.
-    }
+    if (token1 === token2) continue;
+    if (token1 === undefined) return 1;
+    if (token2 === undefined) return -1;
+    if (token1 < token2) return -1;
+    if (token1 > token2) return 1;
   }
+  // Then check verb
+  const verb1 = HTTP_VERBS[route1.verb.toLowerCase()] || HTTP_VERBS.get;
+  const verb2 = HTTP_VERBS[route2.verb.toLowerCase()] || HTTP_VERBS.get;
+  if (verb1 !== verb2) return verb1 - verb2;
   return 0;
```

Comment:
FWIW, this line is not executed by our test suite according to https://coveralls.io/builds/19876890/source?filename=packages/rest/src/router/route-sort.ts

Please consider adding a test to demonstrate what happens when two routes with the same path are configured. (It's probably better to open a new PR for that, so that this patch can be landed ASAP.)